app/models/manageiq/providers/vmware/discovery.rb

```diff
@@ -4,70 +4,61 @@ module ManageIQ
-          rescue => err
-            $log&.debug("Vmware::Discovery: Failed to connect to VMware webservice: #{err}. ip = #{ost.ipaddr}")
-            return
+          # Get Info from VMware (vSphere) Web Service API
+          access_webservice(ost) do |info|
+            if add_hypervisor(ost, info)
+              add_os(ost, info)
+            end
           end
+        rescue StandardError => err
+          _log&.warn("Vmware::Discovery: Failed to connect to VMware webservice: #{err}. ip = #{ost.ipaddr}")
+          return
+        end
 
-          $log&.debug("Vmware::Discovery: ip = #{ost.ipaddr}, Connected to VMware webservice. Machine is either ESX or VirtualCenter.")
-
-          # Next check for ESX or VC. Since VC shares some port numbers with ESX, we check VC before ESX
-
-          # TODO: See if there is a way we can check ESX first, and without having to
-          #   also check VC, since it is more likely there will be more ESX servers on
-          #   a network than VC servers.
+        # Obtains info about IP address from vSphere Web Service API
+        # @param ost [OpenStruct]
+        def self.access_webservice(ost)
+          require 'VMwareWebService/MiqVimClientBase'
+          vim = MiqVimClientBase.new(ost.ipaddr, "test", "test")
+          info = vim&.sic&.about
 
-          checked_vc = false
-          found_vc = false
+          _log&.debug("Vmware::Discovery: ip = #{ost.ipaddr}, Connected to VMware webservice.")
 
-          # Check if we have VC ports
-          if ost.discover_types.include?(:virtualcenter)
-            checked_vc = true
+          yield info
+        end
 
-            if ManageIQ::NetworkDiscovery::Port.all_open?(ost, VC_PORTS)
-              ost.os << :mswin
-              ost.hypervisor << :virtualcenter
-              found_vc = true
-              $log&.debug("Vmware::Discovery: ip = #{ost.ipaddr}, Machine is VirtualCenter.")
-            end
-          end
+        # Adds product type (as hypervisor value) from vmware api info
+        def self.add_hypervisor(ost, info)
+          hypervisor = case info&.productLineId
+                       when 'vpx'
+                         _log&.debug("Vmware::Discovery: ip = #{ost.ipaddr}, Machine is VirtualCenter.")
+                         :virtualcenter
+                       when 'esx'
+                         _log&.debug("Vmware::Discovery: ip = #{ost.ipaddr}, Machine is an ESX server.")
+                         :esx
+                       when 'embeddedEsx'
+                         _log&.debug("Vmware::Discovery: ip = #{ost.ipaddr}, Machine is an ESXi server.")
+                         :esx
+                       when 'gsx'
+                         _log&.debug("Vmware::Discovery: ip = #{ost.ipaddr}, Machine is an VMWare Server product.")
+                         :vmwareserver
+                       else
+                         _log&.error("Vmware::Discovery: ip: #{ost.ipaddr}, Unknown product: #{info&.productLineId}")
```

Comment:
Should be `ip = ...` to stay consistent with the rest of these

Reply:
Why consistency of sentence in log (between different log levels) has bigger priority than given information? Isn't it valuable to know, which unexpected value is inside?

Reply:
ahhh, sorry, you mean  = instead of : ...I noticed after I submitted comment above :)

Reply:
Oh yeah sorry I meant the `:` part, not the rest of the info :)